_ingest-pipelines/processors/text-chunking.md

```diff
@@ -0,0 +1,315 @@
+{
+  "description": "A text chunking and embedding ingest pipeline",
+  "processors": [
+    {
+      "text_chunking": {
+        "algorithm": {
+          "fixed_token_length": {
+            "token_limit": 10,
+            "overlap_rate": 0.2,
+            "tokenizer": "standard"
+          }
+        },
+        "field_map": {
+          "passage_text": "passage_chunk"
+        }
+      }
+    },
+    {
+      "text_embedding": {
+        "model_id": "LMLPWY4BROvhdbtgETaI",
+        "field_map": {
+          "passage_chunk": "passage_chunk_embedding"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+**Step 2 (Optional): Test the pipeline.**
+
+It is recommended that you test your pipeline before you ingest documents.
+{: .tip}
+
+To test the pipeline, run the following query:
+
+```json
+POST _ingest/pipeline/text-chunking-embedding-ingest-pipeline/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex",
+      "_id": "1",
+      "_source":{
+         "passage_text": "This is an example document to be chunked. The document contains a single paragraph, two sentences and 24 tokens by standard tokenizer in OpenSearch."
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The response confirms that in addition to the `passage_text` and `passage_chunk` field, the processor has generated text embeddings for each of the three passages in the `passage_chunk_embedding` field. The embedding vectors are stored in the `knn` field for each chunk:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "testindex",
+        "_id": "1",
+        "_source": {
+          "passage_chunk_embedding": [
+            {
+              "knn": [...]
+            },
+            {
+              "knn": [...]
+            },
+            {
+              "knn": [...]
+            }
+          ],
+          "passage_text": "This is an example document to be chunked. The document contains a single paragraph, two sentences and 24 tokens by standard tokenizer in OpenSearch.",
+          "passage_chunk": [
+            "This is an example document to be chunked. The document ",
+            "The document contains a single paragraph, two sentences and 24 ",
+            "and 24 tokens by standard tokenizer in OpenSearch."
+          ]
+        },
+        "_ingest": {
+          "timestamp": "2024-03-20T03:04:49.144054Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+Once you have created an ingest pipeline, you need to create an index for ingestion and ingest documents into the index. To learn more, see [Step 2: Create an index for ingestion]({{site.url}}{{site.baseurl}}/search-plugins/neural-sparse-search/#step-2-create-an-index-for-ingestion) and [Step 3: Ingest documents into the index]({{site.url}}{{site.baseurl}}/search-plugins/neural-sparse-search/#step-3-ingest-documents-into-the-index) of [Neural sparse search]({{site.url}}{{site.baseurl}}/search-plugins/neural-sparse-search/).
```

Comment:
At the end, let's clarify slightly and lowercase "neural". "of the neural sparse search documentation"?